presentation/src/main/java/me/androidbox/presentation/worker/UploadEventWorker.kt

```diff
@@ -0,0 +1,98 @@
+package me.androidbox.presentation.worker
+
+import android.content.Context
+import android.util.Log
+import androidx.core.app.NotificationCompat
+import androidx.hilt.work.HiltWorker
+import androidx.work.CoroutineWorker
+import androidx.work.ForegroundInfo
+import androidx.work.WorkerParameters
+import androidx.work.workDataOf
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import me.androidbox.data.R
+import me.androidbox.data.local.converter.EventConverter
+import me.androidbox.data.remote.model.request.EventRequestDto
+import me.androidbox.data.remote.network.event.EventService
+import me.androidbox.data.remote.util.CheckResult.checkResult
+import me.androidbox.presentation.util.CreatePhotoMultipart
+import okhttp3.MultipartBody
+import java.util.*
+import kotlin.random.Random
+
+@HiltWorker
+class UploadEventWorker @AssistedInject constructor(
+    @Assisted private val context: Context,
+    @Assisted private val workerParameters: WorkerParameters,
+    private val eventService: EventService,
+    private val eventConverter: EventConverter,
+    private val createPhotoMultipart: CreatePhotoMultipart
+) : CoroutineWorker(context, workerParameters)  {
+
+    override suspend fun doWork(): Result {
+
+        /** TODO Remove this mock data - this is just for testing purposes */
+        val eventRequest = EventRequestDto(
+            id = UUID.randomUUID().toString(),
+            title = "Title",
+            description = "description",
+            from = 3L,
+            to = 6L,
+            remindAt = 4L,
+            attendeeIds = listOf(
+                UUID.randomUUID().toString(),
+                UUID.randomUUID().toString(),
+                UUID.randomUUID().toString(),
+                UUID.randomUUID().toString()
+            )
+        )
+
+        val eventRequestJson = eventConverter.toJson(eventRequest)
+
+        /** TODO FIXME Serialize the event request to json
+        val moshi = Moshi.Builder().build()
+        val jsonAdapter = moshi.adapter<EventRequestDto>(EventRequestDto::class.java)
+        val eventRequestJson = jsonAdapter.toJson(eventRequest)
+         */
+
+        val photoUrl = inputData.getString("photokey") ?: ""
+        val listOfPhotos = listOf(photoUrl)
+
+        /* Create the multipart for the list of photos */
+        val listOfPhotoMultiPart = createPhotoMultipart.createMultipartPhotos(listOfPhotos)
+
+        val responseResult = checkResult {
+            eventService.createEvent(
+                listOfPhoto = listOfPhotoMultiPart,
+                eventBody = MultipartBody.Part.createFormData("create_event_request", eventRequestJson)
+            )
+        }
+
+        /* Check if the request was successful or not */
+        val result = responseResult.fold(
+            onSuccess = { eventDto ->
+                val outputData = workDataOf("eventKey" to eventDto)
+                Result.success(outputData)
```

Comment:
@MKiperszmid @philipplackner Is this correct on how to pass data back from the work manager: I am getting this crash:
```
Work [ id=eb508899-6586-4674-94f7-8a17420983f0, tags={ me.androidbox.presentation.worker.UploadEventWorker } ] failed because it threw an exception/error
    java.util.concurrent.ExecutionException: java.lang.IllegalArgumentException: Key eventKeyhas invalid type class me.androidbox.data.remote.model.response.EventDto
```

Reply:
Another question, what do we do with this EventDto object that is returned from the endpoint? I save the event into Room DB before I sending the request via work manager.

Reply:
Yes it's correct, but you can only pass primitive types (and maybe Parcelables I guess). It just doesn't know how to pass your `EventDto` back as a result. You can try to make it a `Parcelable` and if that doesn't work pass a serialized JSON string of the DTO object.
Either way, I wonder why you need to pass the whole event back? Why not just the info that it was successful?